src/main/room/main/G1HRPS/RoomManager.java

```diff
@@ -1,97 +1,305 @@
+			singleRoomNumber=singleRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				singleVacantCount+=1;
+			}
+			break;	
+		case Standard:
+			standardTotalCount+=1;
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				standardVacantCount+=1;
+				standardRoomNumber=singleRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			}
+			break;
+		case Deluxe:
+			deluxeTotalCount+=1;
+			deluxeRoomNumber=deluxeRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				deluxeVacantCount+=1;
+			}
+			break;
+		case Suite:
+			suiteTotalCount+=1;
+			suiteRoomNumber=suiteRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				suiteVacantCount+=1;
+			}
+			break;
+		case VIP:
+			VIPTotalCount+=1;
+			VIPRoomNumber=VIPRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				VIPVacantCount+=1;
+			}
+			break;
+		default:
+			break;
 
-	/**
-	 * Occupancy rate of each room type.
-	 * e.g. Standard 2/5 occupied, VIP 1/5 occupied
-	 */
-	public void GetRoomStatisticsByTypeOccupancyRate() {
-		// TODO - implement RoomManager.GetRoomStatisticsByTypeOccupancyRate
-		throw new UnsupportedOperationException();
+			
+		}
+	singleRoomNumber =singleRoomNumber.substring(1,singleRoomNumber.length()-1); // removes the last comma in this String
+	standardRoomNumber =standardRoomNumber.substring(1,standardRoomNumber.length()-1);
+	suiteRoomNumber =suiteRoomNumber.substring(1,suiteRoomNumber.length()-1);
+	VIPRoomNumber =VIPRoomNumber.substring(1,VIPRoomNumber.length()-1);
+	deluxeRoomNumber =deluxeRoomNumber.substring(1,deluxeRoomNumber.length()-1);
+	System.out.println("Single: Number: "+ singleVacantCount +" out of "+singleTotalCount+" occupied");
+	System.out.println("        Rooms: "+singleRoomNumber);
+	System.out.println("Standard: Number: "+ standardVacantCount +" out of "+standardTotalCount+" occupied");
+	System.out.println("        Rooms: "+standardRoomNumber);
+	System.out.println("Deluxe: Number: "+ deluxeVacantCount +" out of "+deluxeTotalCount+" occupied");
+	System.out.println("        Rooms: "+deluxeRoomNumber);
+	System.out.println("Suite: Number: "+ suiteVacantCount +" out of "+suiteTotalCount+" occupied");
+	System.out.println("        Rooms: "+suiteRoomNumber);
+	System.out.println("VIP: Number: "+ VIPVacantCount +" out of "+VIPTotalCount+" occupied");
+	System.out.println("        Rooms: "+VIPRoomNumber);
+	
 	}
+}
 
-	/**
-	 * Number of each status.
-	 * e.g. Vacant 5, Occupied 3, Maintenance 4
-	 */
-	public void GetRoomStatisticsByStatus() {
-		// TODO - implement RoomManager.GetRoomStatisticsByStatus
-		throw new UnsupportedOperationException();
+/**
+ * All room numbers, organized by status.
+ * e.g. Vacant : 02-35,02-36,02-38
+ * 		Occupied:02-39,02-61
+ */
+public void GetRoomStatisticsByStatus() {
```

Comment:
Suggest to return a `HashMap<RoomStatus, Int[]>` which will store RoomStatus and an array of room numbers
`e.g. { { RoomStatus.Vacant, { 401, 301, 305 } }, { RoomStatus.Occupied, { 402, 302 } }, { RoomStatus.Maintenance, { } } }`